src/compat.jl

```diff
@@ -16,3 +16,16 @@ function valid_dl_path(path, platform)
         return false
     end
 end
+
+# If we don't have `stdlib_version` from Pkg, recreate it ourselves
+if !isdefined(Pkg.Types, :stdlib_version)
+    function stdlib_version(uuid::Base.UUID, julia_version::Union{VersionNumber,Nothing})::Union{VersionNumber,Nothing}
+        last_stdlibs = Pkg.Types.get_last_stdlibs(julia_version)
```

Comment:
In Julia v1.6 `Pkg.Types.get_last_stdlibs` doesn't return the version numbers, only the names.  Unless there are other ways to get the version number, I believe we're going to drop support for this version